lib/main.dart

```diff
@@ -60,25 +65,50 @@ class _MyHomePageState extends State<MyHomePage> {
   }
 
   void processReceivedData(Uint8List? value) {
-    if ((value == null) || value.isEmpty || (value.length % 3 != 0)) {
+    if (value == null || value.isEmpty || value.length % 15 != 0) {
+      debugPrint('Incorrect buffer size received');
       return;
     }
-    // Decode 24-bit values into integers
-    for (int i = 0; i < value.length; i += 3) {
-      if (i + 2 < value.length) {
-        int intValue = (value[i + 2] << 16) | (value[i + 1] << 8) | value[i];
-        deviceData.add(intValue.toSigned(24));
+
+    for (int packetStart = 0; packetStart < value.length; packetStart += 15) {
+      if (packetStart + 14 < value.length) {
```

Comment:
In my opinion (packetStart + 14 < value.length) cannot happen provided (value.length % 15 == 0)
Debug assert instead of 'if'?